swagger_server/itm/itm_scenario_session.py

```diff
@@ -233,6 +240,87 @@ def _get_sub_directory_names(self, directory):
                 directories.append(item)
         return directories
 
+    def _proper_treatment(self, treatment: str, injury_name: str, location: str) -> bool:
+        # NOTE: Asthmatic, Burns, Forehead Scrape, Ear Bleed are currently untreatable
+
+        """
+            Head Injuries
+            Forehead Scrape: None
+            Face Shrapnel: Airway
+            Ear Bleed: None
+
+            Neck Injuries
+            Neck Puncture: Hemostatic gauze
+
+            Hand Injuries
+            Wrist Amputation: Tourniquet
+            Palm Laceration: Pressure bandage
+
+            Arm Injuries
+            Forearm Laceration: Pressure bandage
+            Bicep Puncture: Tourniquet
+            Shoulder Puncture: Hemostatic gauze
+
+            Chest Injuries
+            Asthmatic: None
+            Chest Collapse: Decompression Needle
+
+            Stomach Injuries
+            Stomach Puncture: Hemostatic gauze
+            Side Puncture: Hemostatic gauze
+
+            Leg Injuries
+            Thigh Puncture: Tourniquet
+            Thigh Laceration: Tourniquet
+            Shin Amputation: Tourniquet
+            Calf Laceration: Pressure bandage
+        """
+        match injury_name:
+            case 'Amputation':
+                return treatment == 'Tourniquet'
+            case 'Chest Collapse':
+                return treatment == 'Decompression Needle'
+            case 'Laceration':
+                if 'thigh' in location:
+                    return treatment == 'Tourniquet'
+                else:
+                    return treatment == 'Hemostatic gauze'
+            case 'Puncture':
+                if 'bicep' in location or 'thigh' in location:
+                    return treatment == 'Tourniquet'
+                else:
+                    return treatment == 'Hemostatic gauze'
+            case 'Shrapnel':
+                return treatment == 'Nasopharyngeal airway'
+            case _:
+                return False
+
+
+    def _reveal_injuries(self, source: Casualty, target: Casualty):
+        if target.assessed: # Don't reveal injuries in assessed casualties
+            return
+        for source_injury in source.injuries:
+            # TBD: Hidden injuries should be configurable by type or injury instance
+            found_burn = False
+            if source_injury.name == 'Burn': # Casualty has a burn injury
+                for target_injury in target.injuries:
+                    if target_injury.name == 'Burn': # Burn has already been revealed
+                        found_burn = True
+                        break
+                if not found_burn:
+                    target.injuries.append(source_injury)
+
+
+    def _clear_hidden_data(self):
+        for casualty in self.scenario.state.casualties:
+            for injury in casualty.injuries:
```

Comment:
We can replace for loop and if statement with this list comprehension to only include injuries that aren't burns:

`casualty.injuries[:] = [injury for injury in casualty.injuries if injury.name != 'Burn']`

Reply:
Done, but used the new `hidden_injury_types` list.